conjure_oxide/tests/generated_tests.rs

```diff
@@ -148,156 +148,195 @@ fn integration_test_inner(
+        println!("Minion solutions: {:#?}", solutions_json);
     }
 
-    assert_eq!(model, expected_model);
+    // Stage 4: Check that the generated rules match expected traces
+    let generated_rule_trace_human = read_human_rule_trace(path, essence_base, "generated")?;
+    let expected_rule_trace_human = read_human_rule_trace(path, essence_base, "expected")?;
+
+    // If ACCEPT = false, we assert all intermediate stages match expected values
+    if !accept {
+        // Check Stage 0 (native parser) if not skipped
+        if config.skip_native_parser != Some(true) {
+            let expected_model = read_model_json(path, essence_base, "expected", "parse")?;
+            let model_native = model_native.expect("model_native should exist here");
+            assert_eq!(model_native, expected_model);
+        }
 
-    // Stage 3: Run the model through the Minion solver and check that the solutions are as expected
-    let solutions = get_minion_solutions(model, 0)?;
+        // Check Stage 1 (parsed model)
+        let expected_model = read_model_json(path, essence_base, "expected", "parse")?;
+        assert_eq!(model, expected_model);
 
-    let solutions_json = save_minion_solutions_json(&solutions, path, essence_base, accept)?;
-    if verbose {
-        println!("Minion solutions: {:#?}", solutions_json)
-    }
+        // Check Stage 2 (rewritten model)
+        let expected_model = read_model_json(path, essence_base, "expected", "rewrite")?;
+        assert_eq!(rewritten_model, expected_model);
 
-    //Stage 4: Check that the generated rules match with the expected in temrs if type, order and count
-    // let generated_rule_trace = read_rule_trace(path, essence_base, "generated", accept)?;
-    // let expected_rule_trace = read_rule_trace(path, essence_base, "expected", accept)?;
+        // Check Stage 3 (solutions)
+        let expected_solutions_json = read_minion_solutions_json(path, essence_base, "expected")?;
 
-    let generated_rule_trace_human =
-        read_human_rule_trace(path, essence_base, "generated", accept)?;
-    let expected_rule_trace_human = read_human_rule_trace(path, essence_base, "expected", accept)?;
+        // Check Stage 4 (human-readable rule trace)
+        assert_eq!(expected_rule_trace_human, generated_rule_trace_human);
 
-    //assert_eq!(expected_rule_trace, generated_rule_trace);
-    assert_eq!(expected_rule_trace_human, generated_rule_trace_human);
+        if verbose {
+            println!("Expected solutions: {:#?}", expected_solutions_json);
+        }
+        assert_eq!(solutions_json, expected_solutions_json);
+
+        save_stats_json(context, path, essence_base)?;
+        return Ok(());
+    }
 
-    // test solutions against conjure before writing
+    // If ACCEPT = true, we skip intermediate assertions and only check conjure solutions
     if accept {
-        let mut conjure_solutions: Vec<BTreeMap<Name, Literal>> =
+        // Check solutions against Conjure
```

Comment:
Pedantic: if we return at the end of the !accept block, we don't need a `if accept {` here, and can get rid of a level of nesting.

(This had me looking to see if there was any code for both at the end, which there doesn't seem to be)